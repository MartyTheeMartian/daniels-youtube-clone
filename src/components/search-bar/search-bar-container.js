import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchBar from './search-bar'
import { setSearchTerm, youtubeSearch } from '../../actions/index'

const mapStateToProps = state => ({
  searchTerm: state.search
})

const mapDispatchToProps = dispatch => bindActionCreators({
  setSearchTerm,
  youtubeSearch }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar)
