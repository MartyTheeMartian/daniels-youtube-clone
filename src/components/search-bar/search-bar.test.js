import React from 'react';
import SearchBar from './search-bar';
import { mapStateToProps, mapDispatchToProps } from './search-bar-container'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('SearchBar', () => {
  it('SearchBar Renders', () => {
    const component = shallow(<SearchBar />)
    expect(component).toMatchSnapshot()
  })
  it('Click on search is called with correct arguments', () => {
    const searchTerm = 'bulldog'
    const youtubeSearch = jest.fn()
    const searchbar = mount(<SearchBar youtubeSearch={ youtubeSearch } searchTerm={searchTerm} />)

    searchbar.find('button').simulate('click')
    expect(youtubeSearch).toHaveBeenCalled()
    expect(youtubeSearch.mock.calls[0][0]).toBe(searchTerm)
  })
  it('Typing will call a set searchTerm', () => {
    const setSearchTerm = jest.fn()
    const searchbar = mount(<SearchBar setSearchTerm={ setSearchTerm } searchTerm={'dog'} />)

    searchbar.find('input').simulate('change', {target: {value:'bulldog'}})
    expect(setSearchTerm).toHaveBeenCalled()
    expect(setSearchTerm.mock.calls[0][0]).toBe('bulldog')
  })
  it('mapStateToProps works correctly', () => {
    const state = {search: 'bulldog'}
    const result = mapStateToProps(state)
    const expected = {searchTerm: 'bulldog'}
    expect(result).toEqual(expected)
  })
})
