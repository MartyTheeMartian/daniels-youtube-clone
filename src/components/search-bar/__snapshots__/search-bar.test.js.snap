// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar Renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Menu
    className="navSearch"
    inverted={true}
>
    <MenuItem
        className="title"
    >
        Daniel's YouTube Clone 
        <Icon
            as="i"
            className="youtube"
            size="large"
        />
    </MenuItem>
    <MenuItem>
        <a
            href="https://github.com/danielmarcgardner/daniels-youtube-clone"
        >
            Checkout The Code 
            <Icon
                as="i"
                className="github"
                size="large"
            />
        </a>
    </MenuItem>
    <MenuItem
        position="right"
    >
        <Input
            className="searchField"
            onChange={[Function]}
            placeholder="Search for your favorite videos here..."
            type="text"
        />
        <Button
            as="button"
            className="searchButton"
            color="red"
            onClick={[Function]}
        >
            Search!
        </Button>
    </MenuItem>
</Menu>,
  "nodes": Array [
    <Menu
      className="navSearch"
      inverted={true}
>
      <MenuItem
            className="title"
      >
            Daniel's YouTube Clone 
            <Icon
                  as="i"
                  className="youtube"
                  size="large"
            />
      </MenuItem>
      <MenuItem>
            <a
                  href="https://github.com/danielmarcgardner/daniels-youtube-clone"
            >
                  Checkout The Code 
                  <Icon
                        as="i"
                        className="github"
                        size="large"
                  />
            </a>
      </MenuItem>
      <MenuItem
            position="right"
      >
            <Input
                  className="searchField"
                  onChange={[Function]}
                  placeholder="Search for your favorite videos here..."
                  type="text"
            />
            <Button
                  as="button"
                  className="searchButton"
                  color="red"
                  onClick={[Function]}
            >
                  Search!
            </Button>
      </MenuItem>
</Menu>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SearchBar />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Menu
          className="navSearch"
          inverted={true}
>
          <MenuItem
                    className="title"
          >
                    Daniel's YouTube Clone 
                    <Icon
                              as="i"
                              className="youtube"
                              size="large"
                    />
          </MenuItem>
          <MenuItem>
                    <a
                              href="https://github.com/danielmarcgardner/daniels-youtube-clone"
                    >
                              Checkout The Code 
                              <Icon
                                        as="i"
                                        className="github"
                                        size="large"
                              />
                    </a>
          </MenuItem>
          <MenuItem
                    position="right"
          >
                    <Input
                              className="searchField"
                              onChange={[Function]}
                              placeholder="Search for your favorite videos here..."
                              type="text"
                    />
                    <Button
                              as="button"
                              className="searchButton"
                              color="red"
                              onClick={[Function]}
                    >
                              Search!
                    </Button>
          </MenuItem>
</Menu>,
        "_debugID": 2,
        "_renderedOutput": <Menu
          className="navSearch"
          inverted={true}
>
          <MenuItem
                    className="title"
          >
                    Daniel's YouTube Clone 
                    <Icon
                              as="i"
                              className="youtube"
                              size="large"
                    />
          </MenuItem>
          <MenuItem>
                    <a
                              href="https://github.com/danielmarcgardner/daniels-youtube-clone"
                    >
                              Checkout The Code 
                              <Icon
                                        as="i"
                                        className="github"
                                        size="large"
                              />
                    </a>
          </MenuItem>
          <MenuItem
                    position="right"
          >
                    <Input
                              className="searchField"
                              onChange={[Function]}
                              placeholder="Search for your favorite videos here..."
                              type="text"
                    />
                    <Button
                              as="button"
                              className="searchButton"
                              color="red"
                              onClick={[Function]}
                    >
                              Search!
                    </Button>
          </MenuItem>
</Menu>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBar />,
}
`;
